src/types/form_logic.ts

```diff
@@ -107,6 +131,57 @@ export type LogicCondition =
   | CategoricalLogicCondition
   | BinaryLogicCondition
   | NumericalLogicCondition
+  | MultiValuedLogicCondition
+
+/**
+ * Types for checkbox logic field
+ */
+// Representation of an option in the logic tab
+export type ClientCheckboxConditionOption = {
+  value: string
+  other: boolean
+}
+
+// Representation of frontend checkbox response after being transformed
+export type ClientLogicCheckboxResponse = Omit<
+  IClientFieldSchema,
+  'fieldValue'
+> & {
+  fieldValue: CheckboxConditionValue
+}
+
+// Representation of backend checkbox response after being transformed
+export type ILogicCheckboxResponse = Omit<ICheckboxResponse, 'answerArray'> & {
+  answerArray: CheckboxConditionValue
+}
+// Representation of a Checkbox condition
+export interface LogicCheckboxCondition
+  extends Omit<IConditionSchema, 'value'> {
+  value: CheckboxConditionValue[]
+}
+// Representation of a checkbox condition value
+export type CheckboxConditionValue = {
+  options: string[]
+  others: boolean
+}
+
+/**
+ * Types needed for logic module inputs
+ */
+// Type for fields before being transformed and passed into the logic module
+export type FieldSchemaOrResponse = IClientFieldSchema | FieldResponse
+
+// Type for client fields after being transformed and passed into the logic module
+export type ILogicClientFieldSchema =
+  | IClientFieldSchema
+  | ClientLogicCheckboxResponse
+
+// Type for backend fields after being transformed and passed into the logic module
+export type LogicFieldResponse =
+  | ISingleAnswerResponse
+  | ILogicCheckboxResponse
+  | ITableResponse
+  | IAttachmentResponse
```

Comment:
does it make things easier if we leave these out? since tables and attachments can't be used in logic anyway